orion/primitives/timeseries_postprocessing.py

```diff
@@ -0,0 +1,180 @@
+import operator
+import numpy as np
+import pandas as pd
+
+ops = {'>': operator.gt,
+       '<': operator.lt,
+       '>=': operator.ge,
+       '<=': operator.le,
+       '==': operator.eq}
+
+def extract_dimension(X, dim=None):
+    """
+    The function checks if the dataset being used is valid i.e has a length greater than 0 and contains the dimension required
+    Args:
+        X (ndarray): 
+            N-dimensional value sequence to iterate over
+        dim (int):
+            Integer indicating the dimension number for a multi-dimensional dataset
+    Returns:
+        ndarray:
+            Returns an nd array that contains a dataset with 2 columns ['timestamp', 'value']
+
+    """
```

Comment:
There are a couple of things to keep in mind when writing docstrings. The most important thing is to be **consistent** with the existing docstrings which should look something like this
```
"""Short description in a single line ending with a dot.

Longer description that can span across multiple lines. Longer
description that can span across multiple lines. Longer description
that can span across multiple lines.

Args:
    arg_name (arg_type):
        argument description.
    arg_name (arg_type):
        Argument description that spans across multiple lines. Argument
        description that spans across multiple lines. Argument description
        that spans across multiple lines.

Returns:
    return_type:
        description of the returned objects
```

Based on the current code, what needs to be fixed is:
* the first sentence is always short
* if the line is longer than 100 characters, break it over multiple lines
* add space between each section (`Args:`, and `Returns:`)
* lastly remove unnecessary white space (line 22)

Applying all these changes would look something like:
```
"""Validate data dimensions.

The function checks if the dataset being used is valid i.e has a length 
greater than 0 and contains the dimension required.

Args:
    X (ndarray): 
        N-dimensional value sequence to iterate over.
    dim (int):
        Integer indicating the dimension number for a multi-dimensional dataset.

Returns:
    ndarray:
        An array that contains a dataset with 2 columns ['timestamp', 'value'].
"""
```

This applies to all the docstrings